safekeeper/src/wal_backup.rs

```diff
@@ -557,8 +562,19 @@ pub async fn delete_timeline(ttid: &TenantTimelineId) -> Result<()> {
     let token = CancellationToken::new(); // not really used
     backoff::retry(
         || async {
-            let files = storage.list_files(Some(&remote_path)).await?;
-            storage.delete_objects(&files).await
+            // Do list-delete in batch_size batches to make progress even if there a lot of files.
+            // Alternatively we could make list_files return iterator, but it is more complicated and
+            // I'm not sure deleting while iterating is expected in s3.
+            loop {
+                let files = storage
+                    .list_files(Some(&remote_path), Some(batch_size))
+                    .await?;
+                if files.is_empty() {
+                    return Ok(()); // done
+                }
+                info!("deleting batch of {} WAL segments", files.len());
```

Comment:
If `list_files` returns sorted files, we can also print `min` and `max` filename in logs, fo visibility.

Reply:
s3 does return sorted, good idea, added